### PR TITLE
Fix #3582 :Merge topic_lessons_title.xml into single xml file

### LINKED ISSUE #3582: Merge topic_lessons_title.xml into single xml file
Currently there are 4 versions of `topic_lessons_title.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is no difference between before and after UI.

### PR DESCRIPTION
**Pull request by shivambh12**
Explanation
Fixes [#3582](https://github.com/oppia/oppia-android/issues/3582)

Essential Checklist

1. The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
2. Any changes to scripts/assets files have their rationale included in the PR explanation.
3. The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
4. The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
5. The PR is made from a branch that's not called "develop" and is up-to-date with "develop".
6. The PR is assigned to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

Before Changes
Phone-port

After Changes
Phone-port(1)

Before Changes
Phone-land

After Changes
Phone-land(1)

Before Changes
Tablet-port

After Changes
Tablet-port(1)

Before Changes
Tablet-land

After Changes
Tablet-land(1)

### CODE REVIEW

**Comment by BenHenning** [comment]
Hi. As of today, some main reviewers have taken time off for the next few weeks, so it may take a little while before we can look at this PR. We appreciate your patience while some of our team members recharge. We'll be fully returning on 4 January 2021.

**Review by Arjupta — COMMENTED** [review]
@shivambh12 the changes you made seems fine. You can ask for another review after resolving the conflicts

**Comment by shivambh12** [comment]
@ARJUPTA How should i remove these conflicts?

**Comment by shivambh12** [comment]
@ARJUPTA How should i remove these conflicts?

**Comment by Arjupta** [comment]
> @ARJUPTA How should i remove these conflicts?

- Checkout to develop locally and pull the latest changes from upstream
- Checkout again to your branch `New2` (not a good branch name) and then merge the develop here.
- Due to conflicts the merge won't happen completely so then resolve those conflicts in Android Studio.
- After resolving all the conflicts stage and commit them.
- Push the changes to `New2` branch (`git push origin New2`) that will resolve the issue.

**Comment by shivambh12** [comment]
@ARJUPTA When I try to pull on main branch. It says that it is up to date. Then also, there are conflicts.

**Comment by Arjupta** [comment]
@kritigupta45 can you help @shivambh12 here to resolve a merge conflict locally.

**Comment by kritigupta45** [comment]
> @kritigupta45 can you help @shivambh12 here to resolve a merge conflict locally.

Sure.

**Comment by shivambh12** [comment]
@ARJUPTA I resolved the issue. Thanks for your time.

**Comment by shivambh12** [comment]
@ARJUPTA PTAL

File: app/src/main/res/values-land/dimens.xml [hunk 1]
```diff
@@ -395,4 +395,9 @@
   
   <!-- StoryFragment -->
   <dimen name="story_fragment_padding_bottom">60dp</dimen>
+
+  <!-- TopicLessonsTitle -->
+  <dimen name="topic_lessons_title_margin_start">100dp</dimen>
+  <dimen name="topic_lessons_title_margin_top">32dp</dimen>
+  <dimen name="topic_lessons_title_margin_end">100dp</dimen>
```
Comment (Arjupta) [inline app/src/main/res/values-land/dimens.xml]: `topic_lessons_title_margin_top` is same for both portrait and landscape so you can remove it from here

File: app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-    <variable
-      name="viewModel"
-      type="org.oppia.android.app.home.recentlyplayed.SectionTitleViewModel" />
-  </data>
-
-  <FrameLayout
-    android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    android:layout_marginTop="32dp">
-
```
Comment (Arjupta) [inline app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml:17]: Maybe you shift this `layout_marginTop` to the text view and use `topic_lessons_title_margin_top` as its value.

**Review by Arjupta — CHANGES_REQUESTED** [review]
@shivambh12 nicely done. Take a look at few nit changes

File: app/src/main/res/values-land/dimens.xml (hunk 1 above)
Comment (shivambh12) [reply app/src/main/res/values-land/dimens.xml]: okk

File: app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml (hunk 2 above)
Comment (shivambh12) [reply app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml:17]: By shifting layout_marginTop to text view. Its layout chnges as shown in screenshot below.
Screenshot 2022-01-16 101247
So,I just changed  32dp with topic_lessons_title_margin_top

**Comment by shivambh12** [comment]
@ARJUPTA PTAL

File: app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml (hunk 2 above)
Comment (Arjupta) [reply app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml:17]: Okay then let it be

**Review by Arjupta — APPROVED** [review]
@shivambh12 can you also fix the PR title to start with `Fix #issue_number :` and also correct the PR description. Other than that the PR looks good to me.

File: app/src/main/res/values-land/dimens.xml (hunk 1 above)
Comment (Arjupta) [reply app/src/main/res/values-land/dimens.xml]: As a reviewer we need to see our comments to verify that the requested changes are done or not. So remember to not resolve them on your own.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**